### DISCUSSION TITLE
3D Brain MRI Images to Fit 4-Channel Input in MONAI Model Zoo

### DISCUSSION THREAD

**Discussion by pnama** [Project-MONAI/MONAI#6978] [Q&A]
I am currently working on a project where I am segmenting cancerous regions from 3D brain MRI scans. I have successfully downloaded and loaded a model from the MONAI Model Zoo (3d brats mri segmentation) to assist with this. However, I encountered a challenge: the model seems to have been trained on 4-channel images, necessitating an input size of 16,4,3,3,3.

My current dataset consists of brain images stored in NIfTI files with a shape of a,b,c. I am seeking guidance on how to appropriately reshape my images to match the model's input requirements without compromising the integrity of the original image structure or information.

Alternatively, would it be more feasible to train a new model myself to better suit the format of my existing data? I am keen to utilize the MONAI model and would appreciate any suggestions on the optimal way to proceed.

**Comment by KumoLiu** [comment]
Hi @pnama,
You can train from scratch based on this [tutorial](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/brats_segmentation_3d.ipynb).
You can also replace the first conv of the model from model zoo to fit your own data.
```
from monai.networks.nets import SegResNet
from monai.networks.blocks.segresnet_block import get_conv_layer
model = SegResNet(
    blocks_down=[1, 2, 2, 4],
    blocks_up=[1, 1, 1],
    init_filters=16,
    in_channels=4,
    out_channels=3,
    dropout_prob=0.2,
)

model.convInit = get_conv_layer(3, 3, 16)
```

Hope it helps, thanks!
